warehouse/forklift/legacy.py

```diff
@@ -413,6 +414,40 @@ def _validate_classifiers(form, field):
             )
 
 
+def _validate_dynamic(_form, field):
+    declared_dynamic_fields = set(map(str.title, field.data or []))
+    disallowed_dynamic_fields = {"Name", "Version", "Metadata-Version"}
+    if invalid := (declared_dynamic_fields & disallowed_dynamic_fields):
+        raise wtforms.validators.ValidationError(
+            f"The following metadata field(s) cannot be marked as dynamic: {invalid!r}",
+        )
+    allowed_dynamic_field = set(DynamicFieldsEnum.enums)
+    if invalid := (declared_dynamic_fields - allowed_dynamic_field):
+        raise wtforms.validators.ValidationError(
+            f"The following values are not valid for Dynamic: {invalid!r}"
+        )
+
+
+_extra_name_re = re.compile("^([a-z0-9]|[a-z0-9]([a-z0-9-](?!--))*[a-z0-9])$")
+
+
+def _validate_extras(form, field):
+    try:
+        metadata_version = packaging.version.Version(form.metadata_version.data)
+    except packaging.version.InvalidVersion as e:
+        raise wtforms.validators.ValidationError(
+            f"The Metadata-Version is invalid: {e}"
+        )
+
```

Comment:
This should already be getting validated elsewhere (and before this function is called) so it should be safe to assume that the metadata version is valid.